plugins/download_statistics/plugin_tests/download_statistics_test.py

```diff
@@ -0,0 +1,146 @@
+#  limitations under the License.
+###############################################################################
+
+
+import os
+import json
+
+from tests import base
+from girder.constants import ROOT_DIR
+
+
+def setUpModule():
+    base.enabledPlugins.append('download_statistics')
+    base.startServer()
+
+
+def tearDownModule():
+    base.stopServer()
+
+
+class DownloadStatisticsTestCase(base.TestCase):
+
+    def setUp(self):
+        base.TestCase.setUp(self)
+
+        # Create admin user
+        admin = {'email': 'admin@email.com',
+                 'login': 'adminLogin',
+                 'firstName': 'adminFirst',
+                 'lastName': 'adminLast',
+                 'password': 'adminPassword',
+                 'admin': True}
+        self.admin = self.model('user').createUser(**admin)
+
+        self.filesDir = os.path.join(ROOT_DIR, 'plugins', 'download_statistics',
+                                     'plugin_tests', 'files')
+
+    def _downloadItem(self, itemId):
+        # Download item through REST api
+        path = '/item/%s/download' % str(itemId)
+        resp = self.request(path, user=self.admin, isJson=False)
+        self.assertStatusOk(resp)
+
+        # Iterate through generator to trigger download events
+        for data in resp.body:
+            data
+
+    def _downloadFile(self, fileId):
+        # Download file through REST api
+        path = '/file/%s/download' % str(fileId)
+        resp = self.request(path, user=self.admin, isJson=False)
+        self.assertStatusOk(resp)
+
+        # Iterate through generator to trigger download events
+        for data in resp.body:
+            data
+
+    def _checkDownloadsCount(self, fileId, count):
+        # Downloads file info and asserts download statistics are accurate
+        path = '/file/' + str(fileId)
+        resp = self.request(path, isJson=True)
+        self.assertStatusOk(resp)
+        data = resp.json
+
+        # The generator is never iterated as to not trigger additional events
+        self.assertEqual(data['downloadStatistics']['started'], count)
+        self.assertEqual(data['downloadStatistics']['requested'], count)
+        self.assertEqual(data['downloadStatistics']['completed'], count)
+
+    def testDownload(self):
+        # Create collection
+        collection = self.model('collection').createCollection('collection1', public=True)
+
+        # Create folder in collection
+        folder = self.model('folder').createFolder(collection, 'folder1',
+                                                   parentType='collection',
+                                                   public=True, )
+        # Create item
+        item = self.model('item').createItem('item1', self.admin, folder)
+
+        # Path to test files
+        file1Path = os.path.join(self.filesDir, 'txt1.txt')
+        file2Path = os.path.join(self.filesDir, 'txt2.txt')
+
+        file1 = ''
```

Comment:
No need to initialize `file1` and `file2` here. Variables declared in a `with` context persist in the scope outside of it.